enclave/core/tracee.c

```diff
@@ -0,0 +1,215 @@
+    if (report_buffer)
+        oe_free(report_buffer);
+#elif defined(_WIN32)
+    // WIN32 support is still under development.
+    // We will have to come back to handle this case
+    ret = true;
+#endif
+
+#else
+// When adding support for non-SGX solutions, we need to find a way to correctly
+// identify whether an enclave is debug-allowed.
+#endif
+    return ret;
+}
+
+/*
+**==============================================================================
+**
+** _handle_oelog_init()
+**
+** Handle the OE_ECALL_LOG_INIT from host and initialize SDK logging
+** configuration
+**
+**==============================================================================
+*/
+oe_result_t _handle_oelog_init(uint64_t arg)
+{
+    oe_result_t result = OE_FAILURE;
+    const char* filename = NULL;
+    oe_log_filter_t* filter = (oe_log_filter_t*)arg;
+    oe_log_filter_t local;
+
+    if (filter == NULL)
+    {
+        result = OE_INVALID_PARAMETER;
+        goto done;
+    }
+
+    if (!oe_is_outside_enclave((void*)filter, sizeof(oe_log_filter_t)))
+    {
+        result = OE_INVALID_PARAMETER;
+        goto done;
+    }
+
+    /* Copy structure into enclave memory */
+    oe_secure_memcpy(&local, filter, sizeof(oe_log_filter_t));
+
+    if (!oe_is_outside_enclave((void*)(local.path), local.path_len))
+    {
+        result = OE_INVALID_PARAMETER;
+        goto done;
+    }
+
+    oe_secure_memcpy(&_active_log_filter, &local, sizeof(oe_log_filter_t));
+    filename = get_filename_from_path(_active_log_filter.path);
+    if (filename)
+    {
+        oe_strlcpy(_enclave_filename, filename, sizeof(_enclave_filename));
+    }
+    else
+    {
+        oe_memset(_enclave_filename, 0, sizeof(_enclave_filename));
+    }
+
+    _debug_allowed_enclave = is_enclave_debug_allowed();
+    result = OE_OK;
+done:
+    return result;
+}
+
+oe_result_t oe_log(log_level_t level, const char* fmt, ...)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_log_args_t* args = NULL;
+    oe_va_list ap;
+    int n = 0;
+    int bytes_written = 0;
+
+    // skip logging for non-debug-allowed enclaves
+    if (!_debug_allowed_enclave)
+    {
+        result = OE_OK;
+        goto done;
+    }
+
+    // Check if this message should be skipped
+    if (level > _active_log_filter.level)
+    {
+        result = OE_OK;
+        goto done;
+    }
+    // Validate input
+    if (!fmt)
+    {
+        result = OE_INVALID_PARAMETER;
+        goto done;
+    }
+
+    // Prepare a log record for sending to the host for logging
+    if (!(args = oe_host_malloc(sizeof(oe_log_args_t))))
+    {
+        result = OE_OUT_OF_MEMORY;
+        goto done;
+    }
+
+    bytes_written = oe_snprintf(
+        args->message, OE_LOG_MESSAGE_LEN_MAX, "%s:", _enclave_filename);
+
+    if (bytes_written < 0)
+        goto done;
+
+    args->level = level;
+    oe_va_start(ap, fmt);
+    n = oe_vsnprintf(
+        &args->message[bytes_written],
+        OE_LOG_MESSAGE_LEN_MAX - oe_strlen(_enclave_filename),
```

Comment:
This should be _OE_LOG_MESSAGE_LEN_MAX -  bytes_written_ since that many bytes of the message buffer has been used.

Reply:
addressed